JsonRead/src/main/java/weather/ReadFromInternet.java

```diff
@@ -0,0 +1,19 @@
+package weather;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.function.Function;
+
+public class ReadFromInternet implements ReadFrom {
+
+    public WeatherForecast readFrom(String str) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
You do not need to create ObjectMapper every time you want to readFrom(). It should be a dependency (field) of ReadFromInternet class, created either in the constructor or passed as a parameter to constructor (Inversion of Control).